app/models/grda_warehouse/client_contact.rb

```diff
@@ -55,5 +55,31 @@ def full_address
       @full_address << "Address: #{address_or_note}" if address_or_note.present?
       @full_address.compact.join("\n")
     end
+
+    def self.as_health_contacts(_force = false)
+      contact_columns = {
+        source_id: :id,
+        category: :contact_type,
+        email: :email,
+        collected_on: :last_modified_at,
+      }.invert
+
+      patient_ids = Health::Patient.bh_cp.pluck(:client_id, :id).to_h
+
+      joins(:client).
+        merge(GrdaWarehouse::Hud::Client.
+          full_housing_release_on_file.
+          where(id: Health::Patient.bh_cp.pluck(:client_id))).
```

Comment:
I think probably `find_each` would be helpful.  The `contact` array won't be that big, but you might end up with 25k+ client contact records being instantiated.